runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/graph/LengthPrefixUnknownCodersTest.java

```diff
@@ -252,11 +253,11 @@ public void testLengthPrefixAndReplaceForRunnerNetwork() {
       }
     }
 
-    Set expectedInstructions =
+    ImmutableSet<GenericJson> expectedInstructions =
         ImmutableSet.of(
-            prefixedReadNode.getParallelInstruction(), prefixedReadNodeOut.getInstructionOutput());
+            prefixedReadNodeOut.getInstructionOutput(), prefixedReadNode.getParallelInstruction());
```

Comment:
JSON does not have a set. Comparing list requires ordering. Guava's ImmutableSet keeps ordering. https://github.com/google/guava/wiki/ImmutableCollectionsExplained#how

Reply:
The intent isn't to have a set in JSON but to match the expected vs the received which is why set was being used instead of a list so this test was attempting to do something like `assertThat(..., contains(...))`.

Please update this so that we don't care about the order of the nodes when matching.

Reply:
Added `assertThat(..., containsInAnyOrder(...))` by introducing a new matcher.